src/native/public/monoapi.cmake

```diff
@@ -138,3 +138,23 @@ if(INSTALL_MONO_API)
   install(FILES ${jit_public_headers} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mono-2.0/mono/jit)
   install(FILES ${jit_public_headers_details} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mono-2.0/mono/jit/details)
 endif()
+
+
+# create a helper source file to validate an implementation exists for each public API,
+# it is linked into mono-sgen and will cause a linking failure there if an API is missing
+foreach(header IN LISTS jit_public_headers metadata_public_headers utils_public_headers)
+	if(NOT header MATCHES "profiler-events.h") # profiler-events.h doesn't define APIs which causes an issue
+		string(APPEND publicapis_headers_include "#include <${header}>\n")
+	endif()
+endforeach()
+
+foreach(header IN LISTS jit_public_headers_details metadata_public_headers_details utils_public_headers_details)
+	if(header MATCHES "-functions.h")
+		string(APPEND publicapis_headers_details_include "#include <${header}>\n")
+	endif()
+endforeach()
+
+configure_file(validate-apis.c.in ${CMAKE_CURRENT_BINARY_DIR}/validate-apis.c @ONLY)
+
+add_library(monoapi-validate INTERFACE)
+target_sources(monoapi-validate INTERFACE ${CMAKE_CURRENT_BINARY_DIR}/validate-apis.c)
```

Comment:
I don't think this is an interface library.  I think it's STATIC?

Reply:
I guess it doesn't matter in this case.  making it an interface library just  means we will build it from source instead of linking from a .a.  but we only use it in one target anyway.

Reply:
thanks, yeah I just copied it from above, but looks like interface libraries with sources are [new in CMake 3.19](https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries):
> New in version 3.19: An interface library target may be created with source files:

So that wouldn't work anyway since we're targetting an earlier CMake.

I've switched to just defining the source here and including it directly into mono-sgen.

Reply:
> I've switched to just defining the source here and including it directly into mono-sgen.

😢 I kind of wish we'd stop doing that. but I really can't articulate why.  feels like cmake really wants you to use static or object libraries for sharing chunks of the build from place to place.

but incrementally this is just one more use of that pattern that we have all over mini/ so I guess it's fine.